### DISCUSSION TITLE
ModuleNotFoundError: No module named 'triqs'

### DISCUSSION THREAD

**Discussion by AQuantumGuy** [TRIQS/triqs#848] [Q&A] [answered]
Hi,
I'm completely new to this field and I was trying to download TRIQS on my linux virtual machine. I followed the tutorial from the website "https://triqs.github.io/triqs/latest/install.html" and in specific from the "Ubuntu Debian packages" section. I successfully downloaded, but when I try to run a triqs code (import) on jupyter notebook it says "ModuleNotFoundError: No module named 'triqs'". Could anyone let me know how to import the module on it?

**Comment by the-hampel** [comment, marked as answer]
Hi, 
In principle this should work just fine. So you are running the jupyter notebook directly in the VM and installed the package via `apt-get install`? Could you try to run in the VM on the command line: 
```
python3 -c "import triqs; print(triqs.__file__)"
```
Best,
the-hampel

P.S. If you happen to be in Canada, Sherbrooke right now. Please join us at the bar. ;-)

**Reply by AQuantumGuy** [reply]
Hi,
Just tried running that and it gives:

_Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'triqs'_

Basically, I'm using VM with Quantum Espresso. TRIQS works if I install it through docker on commmand line "docker pull flatironinstitute/triqs". But I wanted to install the TRIQS solid_dmft package as well and since that package does not have the "docker" option, I have to install using the "https://flatironinstitute.github.io/solid_dmft/install.html" instructions but again there I get the error of no module found, so I wanted to fix the issue of installing TRIQS normally.


PS: Thanks for the invite, but I'm from London, UK   :)

**Reply by the-hampel** [reply]
Something seems to be not right with the installation. Did you activate inside of the VM a special environment via conda or pipenv? If you check the output of 
```
apt list --installed
```
is the triqs package in the list? can you check if the command 
```
which triqs++
```
and see if the system finds basic parts of triqs?

**Reply by AQuantumGuy** [reply]
In the whole list I found these two:

triqs/unknown,now 3.0.2 amd64 [installed]
triqs_dft_tools/unknown,now 3.0.0 amd64 [installed]

**Reply by AQuantumGuy** [reply]
Also, what do you mean by "Did you activate inside of the VM a special environment via conda or pipenv?" because I just managed to install and not setting something extra. 

(And apologies if I sound very beginner, just learning this for my PhD)

**Reply by the-hampel** [reply]
that looks okay. Do you use inside of the VM any special environment? Like Conda or pipenv?

**Reply by the-hampel** [reply]
I think the problem might be related to something else in your VM. The installation of the package looks okay. Can you show me what 
```
which python3
```
gives you? 

Also I have prepared a docker file that has all relevant triqs applications including solid_dmft and wannier09 installed: https://hub.docker.com/repository/docker/materialstheory/w90_school_triqs

**Reply by AQuantumGuy** [reply]
"which python3" gave me "/usr/bin/python3" only

**Reply by the-hampel** [reply]
That looks perfectly fine. This should be the system python, which should be able to see triqs. Can you see if you find a folder of triqs in: 
```
/usr/lib/python3.9/site-packages/
```
I am not sure if you have python 3.9. So maybe you have to look instead into similar folder, e.g. `python3.8` or so. 

- Alex

**Reply by AQuantumGuy** [reply]
I have tried looking into python folders, I have python 2.7, 3, 3.6, 3.7, 3.8, but could not find triqs folder in any of them

**Reply by AQuantumGuy** [reply]
Update: I found triqs folder in the following path "/usr/lib/python3.6/dist-packages/triqs/"

**Reply by AQuantumGuy** [reply]
When I run the command   " python3 -c "import triqs; print(triqs.__file__)"  , I get    /usr/lib/python3.6/dist-packages/triqs/__init__.py

**Reply by the-hampel** [reply]
Ah okay. So that mean it works now? Because your command did not give an error?

**Reply by AQuantumGuy** [reply]
It actually gave an error before as I did not have a triqs folder. The issue I believe is not with TRIQS but with my VM. I still can't import triqs on Jupyter Notebook though

**Reply by the-hampel** [reply]
Actually this problem is not really related to triqs I feel. The problem is more that you have many versions of python installed in your VM and triqs lives in python3.6. But your jupyter kernel is probably at a different version. Can you do in a jupyter notebook where triqs does not load: 
```
import sys
print(sys.version)
```
And check if it is using 3.6? Probably not?

**Reply by AQuantumGuy** [reply]
Just done that and on Jupyter Notebook I have Python 3.7.13

**Reply by the-hampel** [reply]
Yes so this is the reason why you are not seeing triqs. Your jupyter kernel is running Python 3.7 and searching inside of `/usr/lib/python3.7/dist-packages` for extra packages. But triqs is installed in python3.6. I am not sure how this happened. Can you point me to the VM you are using? It is hard to understand the setup of your VM without further knowledge about the VM itself.

**Reply by AQuantumGuy** [reply]
Hi, 
Thank you so much. Now it works! The issue was that I had a different verion of Python running on Jupyter Notebook. Fixed that by writing on the command line the following command:

`python3.6 -m ipykernel install --user`

Having done that, when I tried to import triqs on Jupyter Notebook it worked!! Now I'll try and install other TRIQS packages which I need (Solid_DMFT and DFTTools). I believe it should install easily with the fixed error.
I was using Oracle VM and was running Quantum Mobile on it (as I need Quantum Espresso) and you can find in the following link: 
https://quantum-mobile.readthedocs.io/en/latest/

Thank again for your help, you was really helpful with prompt replies. Truely appreciate it!

**Reply by AQuantumGuy** [reply]
I'm actually trying to follow the Wannier-tutorials from 2022_05_Trieste day. I saw that you created all those codes as well, so hopefully will get back to you on that as well. 

Not sure if I shall as here or in a different post, but how can one run following command which is on the tutorial (https://github.com/wannier-developers/wannier-tutorials/blob/master/2022_05_Trieste/DAY3_AM_TRIQS/triqs_tutorial_solution.ipynb)?

ssh your@mail.com@ins45124.ictp.it -L 7899:localhost:8888 -L 7898:localhost:8889

I mean what shall I replace in "your@mail.com@ins45124.ictp.it"

**Reply by the-hampel** [reply]
Perfect. Yes that is a solution. Was not sure if that clashes with other things on your system. Just be careful when installing other python packages. They need to be installed with this python3.6 to work in the same jupyter notebook. 

Regarding your second question: You don't need to do that since you are already on the VM. Just ignore that part.

**Reply by AQuantumGuy** [reply]
Great, thanks!! Looking forward to work on this now! Thanks again for your help!
